addons/website_form/static/src/snippets/s_website_form/options.js

```diff
@@ -1,635 +1,1280 @@
+            case 'type_opt':
+                return !this.$target[0].classList.contains('o_website_form_model_required');
         }
-    });
-
-    // radio / checkboxes field option to display them horizontally or vertically
-    options.registry['website_form_editor_choice_field_display'] = options.Class.extend({
-        xmlDependencies: ['/website_form/static/src/xml/website_form_editor.xml'],
-
-        /**
-         * @see this.selectClass for parameters
-         */
-        website_form_choice_field_display: function (previewMode, widgetValue, params) {
-            this.$target.toggleClass('o_website_form_flex_fw', widgetValue === 'vertical');
-            this.$target[0].dataset.display = widgetValue;
-        },
-
-        //----------------------------------------------------------------------
-        // Private
-        //----------------------------------------------------------------------
-
-        /**
-         * @override
-         */
-        _computeWidgetState: function (methodName, params) {
-            if (methodName === 'website_form_choice_field_display') {
-                return this.$target.attr('data-display');
+        return this._super(...arguments);
+    },
+    /**
+     * @override
+     */
+    _renderCustomXML: function (uiFragment) {
+        const selectEl = uiFragment.querySelector('we-select[data-name="type_opt"]');
+        const fieldType = this._getFieldType();
+        const fieldsInForm = Array.from(this.formEl.querySelectorAll('.s_website_form_field:not(.o_website_form_custom)')).map(field => field.dataset.type).filter(el => el !== fieldType);
+        this.existingFields.forEach(option => {
+            if (!fieldsInForm.includes(option.dataset.existingField)) {
+                selectEl.append(option.cloneNode(true));
             }
-            return this._super(...arguments);
-        },
-    });
-
-    // Generic custom field options
-    options.registry['website_form_editor_field'] = options.Class.extend({
-        xmlDependencies: ['/website_form/static/src/xml/website_form_editor.xml'],
-
-        // Option to toggle inputs required attribute
-        website_form_field_require: function (previewMode, widgetValue, params) {
-            this.$target.find('.o_website_form_input').each(function (index, input) {
-                input.required = !input.required;
+        });
+    },
+    /**
+     * Replace the target content with the field provided
+     *
+     * @private
+     * @param {Object} field
+     * @returns {Promise}
+     */
+    _replaceField: function (field) {
+        return this._renderField(field).then((htmlField) => {
+            // Replace the target with the new field without removing it from the dom.
+            this.$target.html(htmlField.innerHTML);
+            this.$target[0].classList = htmlField.classList;
+            htmlField.dataset.type ? this.$target[0].dataset.type = htmlField.dataset.type : delete this.$target[0].dataset.type;
```

Comment:
That is Ugly... but flickering I hate it ! What can I do to replace my target without reloading my widget ?

Reply:
My comment was for the use of ternary form for non-assignment instructions.
But now that you tell it, is not there a way to use DOM node moves instead of HTML replacement? It would make odony happy.